### PR TITLE
don't set PROXY_XDS_VIA_AGENT in multicluster mode

### PR DESCRIPTION
**Pull request by stevenctl**
Haven't fully looked into why this breaks things yet, but it causes ingress gateway in remote clusters to not come up during multicluster setup. Their logs have:

```
2020-09-01T18:24:40.785873Z	info	xdsproxy	connecting to upstream istiod-remote.istio-system.svc:15012
2020-09-01T18:24:40.801011Z	error	xdsproxy	upstream recv error: rpc error: code = Unknown desc = authentication failure
2020-09-01T18:24:40.801140Z	error	xdsproxy	downstream recv error: rpc error: code = Canceled desc = context canceled
2020-09-01T18:24:40.801490Z	warning	envoy config	StreamAggregatedResources gRPC config stream closed: 2, authentication failure
2020-09-01T18:24:54.663487Z	warn	Envoy proxy is NOT ready: config not received from Pilot (is Pilot running?): cds updates: 0 successful, 0 rejected; lds updates: 0 successful, 0 rejected
```

After this was added, the job started failing: https://prow.istio.io/?job=integ-pilot-multicluster-tests_istio

```
✔ Istiod installed
- Processing resources for Ingress gateways.
- Processing resources for Ingress gateways. Waiting for Deployment/istio-system/istio-ingressgat...
✔ Istiod installed
- Processing resources for Ingress gateways.
- Processing resources for Ingress gateways. Waiting for Deployment/istio-system/istio-ingressgat...
✘ Ingress gateways encountered an error: failed to wait for resource: resources not ready after 5m0s: timed out waiting for the condition
Deployment/istio-system/istio-ingressgateway
- Pruning removed resources
✘ Ingress gateways encountered an error: failed to wait for resource: resources not ready after 5m0s: timed out waiting for the condition
Deployment/istio-system/istio-ingressgateway
- Pruning removed resources2020-09-01T18:29:21.427118Z	info	tf	=== FAILED: Deploy Istio [Suite=pilot] ===
2020-09-01T18:29:21.427186Z	error	tf	=== Dumping Istio Deployment State...
2020-09-01T18:29:26.464652Z	error	tf	Test setup error: 2 errors occurred deploying remote clusters: 2 errors occurred:
	* failed deploying control plane to cluster cluster-4: install failed: failed to install manifests: errors occurred during operation
	* failed deploying control plane to cluster cluster-1: install failed: failed to install manifests: errors occurred during operation

```

cc @rshriram 

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[X] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[X] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[X] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: tests/integration/telemetry/tracing/clienttracing/client_tracing_test.go [hunk 1]
```diff
@@ -18,6 +18,7 @@ package client
 import (
 	"errors"
 	"fmt"
+	"istio.io/istio/pkg/test/framework/resource"
```
Comment (bianpengyuan) [inline tests/integration/telemetry/tracing/clienttracing/client_tracing_test.go:21]: The position of this new import looks weird to me, same happens to several other files.

Comment (ericvn) [reply tests/integration/telemetry/tracing/clienttracing/client_tracing_test.go:21]: I thought we had or where putting in code changes to handle import ordering. Maybe we aren't testing PRs yet.

Comment (howardjohn) [reply tests/integration/telemetry/tracing/clienttracing/client_tracing_test.go:21]: we should... looking into it

**Comment by stevenctl** [comment]
>  I did not realize that the multi-cluster test used the same pilot/main_test.go..

Fully converting it is a WIP.

File: tests/integration/telemetry/tracing/clienttracing/client_tracing_test.go (hunk 1 above)
Comment (howardjohn) [reply tests/integration/telemetry/tracing/clienttracing/client_tracing_test.go:21]: https://github.com/golangci/golangci-lint/issues/1351

Comment (bianpengyuan) [reply tests/integration/telemetry/tracing/clienttracing/client_tracing_test.go:21]: @stevenctl  Looks like this import and the one in tracing_test.go is still not fixed?

Comment (stevenctl) [reply tests/integration/telemetry/tracing/clienttracing/client_tracing_test.go:21]: Thanks

File: tests/integration/pilot/main_test.go [hunk 2]
```diff
@@ -103,10 +103,13 @@ func TestMain(m *testing.M) {
 			}
 			return ctx.Config().ApplyYAML("", string(crd))
 		}).
-		Setup(istio.Setup(&i, func(cfg *istio.Config) {
+		Setup(istio.Setup(&i, func(ctx resource.Context, cfg *istio.Config) {
 			cfg.Values["telemetry.v2.metadataExchange.wasmEnabled"] = "false"
 			cfg.Values["telemetry.v2.prometheus.wasmEnabled"] = "false"
-			cfg.Values["meshConfig.defaultConfig.proxyMetadata.PROXY_XDS_VIA_AGENT"] = "enable"
+			if !ctx.Clusters().IsMulticluster() {
+				// TODO make this compatible with multicluster
```
Comment (incfly) [inline tests/integration/pilot/main_test.go:110]: TODO(stevenctl):

Comment (incfly) [reply tests/integration/pilot/main_test.go:110]: also might worth commenting on why we can't set this field when it's not running as multi cluster mode?

**Pull request merged**